### PR TITLE
Fix `find_chr_next_occurance_simd`

### PR DESCRIPTION
**Pull request by soraros**

### CODE REVIEW

**Comment by MoSafi2** [comment]
Bug fix added 25% to the perofmance.
Great addition @soraros.

**Comment by soraros** [comment]
@MoSafi2 That's rather unexpected, as this is only meant to fix a logical error.

File: helpers.mojo [hunk 1]
```diff
@@ -19,22 +27,18 @@ fn find_chr_next_occurance_simd[
...
-            let bool_vec = simd_vec == chr
-            if bool_vec.reduce_or():
-                for i in range(len(bool_vec)):
-                    if bool_vec[i]:
-                        return start + nv + i
-
-    for n in range(max_iter * simd_width, t_slice):
-        let simd_vec = in_tensor.simd_load[1](start + n)
-        if simd_vec == chr:
-            return start + max_iter * simd_width + n
```
Comment (soraros) [inline helpers.mojo:37]: Should have been `start + n` I think.

Comment (MoSafi2) [reply helpers.mojo:37]: I think here is the reason why the function was returning garbage at the end of the tensors.

**Comment by MoSafi2** [comment]
@soraros It seems that you are right, I tried replacing your function with the old one, and there were no change in performance (still 25% better than yesterday). I did some minor fixes today in the record_trim() function, I will check if one of them caused the increase and I will add tests.

**Comment by soraros** [comment]
I think you can now also remove the workaround in `get_next_line`.

**Comment by MoSafi2** [comment]
@soraros yes, but I also tested that and it added only small gains in performance.

**Comment by MoSafi2** [comment]
@soraros the speed up came from fixing another bug today, this line from [trim_record()](https://github.com/MoSafi2/MojoFastTrim/blob/2fba223704080bbabf69c2e4aa490e1629f482a0/fastq_record.mojo#L86) function was adding extra unecessary calls to ord() and it also produces wrong results. removing the ord() calls causes almost all the perfomance gains.
previous:
`s += (ord(self.QuStr[n - i]) - 33) - quality_threshold
`
now:
` s += self.QuStr[n - i] - 33 - quality_threshold`

**Pull request merged**